### ISSUE TITLE
500 For a PR

### ISSUE THREAD

**Issue by jag3773** [unfoldingWord-dev/gogs#102]
A regular user can't create a PR against https://git.door43.org/Door43/en-uhg.

Sanitized log:
```
Nov 30 02:00:00 git gogs: [Macaron] 2016-11-30 02:00:00: Started POST /Door43/en-uhg/compare/master...danielfrankbowman:master for [::1]
Nov 30 02:00:00 git gogs: 2016/11/30 02:00:00 #033[1;36m[T] PullRequest[0].testPatch (patchPath): /home/git/gogs-repositories/door43/en-uhg.git/pulls/5.patch#033[0m
Nov 30 02:00:00 git gogs: 2016/11/30 02:00:00 #033[1;31m[...routers/repo/pull.go:710 CompareAndPullRequestPost()] [E] NewPullRequest: testPatch: UpdateLocalCopy: git checkout master: exit status 1 - error: you need to resolve your current index first
Nov 30 02:00:00 git gogs: 2016/11/30 02:00:00 #033[1;34m[D] Template: status/500#033[0m
Nov 30 02:00:00 git gogs: [Macaron] 2016-11-30 02:00:00: Completed /Door43/en-uhg/compare/master...danielfrankbowman:master 500 Internal Server Error in 303.215875ms
```

Maybe related to #101.

See if this can be reproduced on latest code.  If it can, create new issue to fix the problem.  Otherwise, retest on production after #100 .

**Comment by richmahn** [comment]
Unable to reproduce with the same repo copied over to test.door43.org:3000 and then forked to my user account, make a change, and then make a PR. I was able to make it happen if I did it on production. Hopefully will be fixed in the next update of Gogs.
